### PR TITLE
Encode the nextRefreshDate to UserDefaults

### PR DESCRIPTION
**Pull request by rodericj**
There was a bug where the nextRefreshDate was not being persistent between encode/decodes. This  bridges that gap.

### CODE REVIEW

File: Common/Operations/Nightscout.swift [hunk 1]
```diff
@@ -329,7 +328,7 @@ public extension Site {
                 updatedSite.deviceStatuses = deviceStatus
             }
             
-            updatedSite.lastUpdatedDate = Date()
+            updatedSite.nextRefreshDate = Date()
```
Comment (someoneAnyone) [inline Common/Operations/Nightscout.swift]: Does it make sense to still call this "nextRefreshDate"? We're putting the date it was last updated here. Also... should we calculate the nextRefreshDate from the last SVG record we received?

File: Common/Parsers/JSONParser.swift [hunk 2]
```diff
@@ -45,7 +45,17 @@ extension Site: Encodable, Decodable {
         let encodedDeviceStatus: [[String : Any]] = deviceStatuses.flatMap{ $0.encode() }
         let encodedComplicationTimeline: [[String : Any]] = complicationTimeline.flatMap { $0.encode() }
         
-        return [JSONKey.url : url.absoluteString, JSONKey.overrideScreenLock : overrideScreenLock, JSONKey.disabled: disabled, JSONKey.uuid: uuid.uuidString, JSONKey.configuration: configuration?.encode() ?? "", JSONKey.sgvs: encodedSgvs, JSONKey.cals: encodedCals, JSONKey.mbgs: encodedMgbs, JSONKey.deviceStatuses: encodedDeviceStatus, JSONKey.complicationTimeline: encodedComplicationTimeline]
+        return [JSONKey.url : url.absoluteString,
+                "nextRefreshDate" : nextRefreshDate,
```
Comment (someoneAnyone) [inline Common/Parsers/JSONParser.swift]: Ok, we should added this to the JSONKey struct.

File: Common/Parsers/JSONParser.swift [hunk 3]
```diff
@@ -469,155 +484,6 @@ extension ComplicationTimelineEntry: Encodable, Decodable {
...
- // makes sure things are sorted correctly by date. When delta's come in they might screw up the order.
- self.sgvs.sorted()
- self.cals.sorted()
- self.mbgs.sorted()
- self.deviceStatuses.sorted()
- 
- }
- }
- */
-
```
Comment (someoneAnyone) [inline Common/Parsers/JSONParser.swift:488]: sounds good. This was old parsing code for socket.io data. We are still using REST.

File: Common/Operations/Nightscout.swift [hunk 4]
```diff
@@ -119,33 +119,32 @@ public class NightscoutDownloader {
         // 2. Provide the API key, passphrase or api-secret token. User of the api prvides a string and this will convert to a SHA1 string.
         headers["api-secret"] = apiSecret?.sha1()
         
-        var requstURL: URL
```
Comment (someoneAnyone) [inline Common/Operations/Nightscout.swift:122]: Sorry about that!

**Review by someoneAnyone — COMMENTED** [review]
Thanks for doing this! I'm making a testable build now.

File: Common/Operations/Nightscout.swift (hunk 4 above)
Comment (rodericj) [reply Common/Operations/Nightscout.swift:122]: Not a problem. Just cleaning up where possible.

File: Common/Operations/Nightscout.swift (hunk 1 above)
Comment (rodericj) [reply Common/Operations/Nightscout.swift]: Good points. I'll revisit now.

File: Common/Parsers/JSONParser.swift (hunk 2 above)
Comment (rodericj) [reply Common/Parsers/JSONParser.swift]: Sure, though I feel like these aren't really JSON keys at this point. It's somewhat misleading.

Comment (someoneAnyone) [reply Common/Parsers/JSONParser.swift]: True.

**Pull request merged**